src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/ExtendedCompletionItemKinds.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor.LanguageServer.Common;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Completion
+{
+    internal enum ExtendedCompletionItemKinds
+    {
+        None = 0,
+        Text = CompletionItemKind.Text,
+        Method = CompletionItemKind.Method,
+        Function = CompletionItemKind.Function,
+        Constructor = CompletionItemKind.Constructor,
+        Field = CompletionItemKind.Field,
+        Variable = CompletionItemKind.Variable,
+        Class = CompletionItemKind.Class,
+        Interface = CompletionItemKind.Interface,
+        Module = CompletionItemKind.Module,
+        Property = CompletionItemKind.Property,
+        Unit = CompletionItemKind.Unit,
+        Value = CompletionItemKind.Value,
+        Enum = CompletionItemKind.Enum,
+        Keyword = CompletionItemKind.Keyword,
+        Snippet = CompletionItemKind.Snippet,
+        Color = CompletionItemKind.Color,
+        File = CompletionItemKind.File,
+        Reference = CompletionItemKind.Reference,
+        Folder = CompletionItemKind.Folder,
+        EnumMember = CompletionItemKind.EnumMember,
+        Constant = CompletionItemKind.Constant,
+        Struct = CompletionItemKind.Struct,
+        Event = CompletionItemKind.Event,
+        Operator = CompletionItemKind.Operator,
+        TypeParameter = CompletionItemKind.TypeParameter,
+
+        // Kinds custom to VS, starting with index 118115 to avoid collisions with other clients's custom kinds.
```

Comment:
Are we guaranteed a unique range of values or was `118115` chosen randomly?

Reply:
118115 was purposefully chosen randomly to not conflict with other clients who expand the list.